lib/matplotlib/_constrained_layout.py

```diff
@@ -13,58 +13,49 @@
 See Tutorial: :doc:`/tutorials/intermediate/constrainedlayout_guide`
 """
 
-# Development Notes:
-
-# What gets a layoutbox:
-#  - figure
-#    - gridspec
-#      - subplotspec
-#        EITHER:
-#         - axes + pos for the axes (i.e. the total area taken by axis and
-#            the actual "position" argument that needs to be sent to
-#             ax.set_position.)
-#           - The axes layout box will also encompass the legend, and that is
-#             how legends get included (axes legends, not figure legends)
-#         - colorbars are siblings of the axes if they are single-axes
-#           colorbars
-#        OR:
-#         - a gridspec can be inside a subplotspec.
-#           - subplotspec
-#           EITHER:
-#            - axes...
-#           OR:
-#            - gridspec... with arbitrary nesting...
-#      - colorbars are siblings of the subplotspecs if they are multi-axes
-#        colorbars.
-#   - suptitle:
-#      - right now suptitles are just stacked atop everything else in figure.
-#        Could imagine suptitles being gridspec suptitles, but not implemented
-#
-#   Todo:    AnchoredOffsetbox connected to gridspecs or axes.  This would
-#        be more general way to add extra-axes annotations.
-
 import logging
 
 import numpy as np
 
 import matplotlib.cbook as cbook
-import matplotlib._layoutbox as layoutbox
 
 _log = logging.getLogger(__name__)
 
-
-def _spans_overlap(span0, span1):
-    return span0.start in span1 or span1.start in span0
-
-
-def _axes_all_finite_sized(fig):
-    """Return whether all axes in the figure have a finite width and height."""
-    for ax in fig.axes:
-        if ax._layoutbox is not None:
-            newpos = ax._poslayoutbox.get_rect()
-            if newpos[2] <= 0 or newpos[3] <= 0:
-                return False
-    return True
+"""
+General idea:
```

Comment:
vaguely prefer this to be a comment rather than a string...

Reply:
You mean w/ a bunch of hashes in front?